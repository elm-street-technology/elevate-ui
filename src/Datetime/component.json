{
  "description": "A Date/Time component for use in forms. The <Datetime /> component is a styled wrapper around React-Datetime.",
  "displayName": "Datetime",
  "methods": [],
  "props": {
    "classes": {
      "flowType": {
        "name": "Object"
      },
      "required": true,
      "description": ""
    },
    "className": {
      "flowType": {
        "name": "string"
      },
      "required": true,
      "description": "classNames to be passed to the component."
    },
    "field": {
      "flowType": {
        "name": "Object"
      },
      "required": true,
      "description": ""
    },
    "form": {
      "flowType": {
        "name": "Object"
      },
      "required": true,
      "description": ""
    },
    "id": {
      "flowType": {
        "name": "string"
      },
      "required": true,
      "description": ""
    },
    "label": {
      "flowType": {
        "name": "string"
      },
      "required": true,
      "description": "Text input to be used as the label for the component."
    },
    "theme": {
      "flowType": {
        "name": "Object"
      },
      "required": true,
      "description": ""
    },
    "withScaffold": {
      "flowType": {
        "name": "boolean"
      },
      "required": false,
      "description": "",
      "defaultValue": {
        "value": "true",
        "computed": false
      }
    }
  }
}