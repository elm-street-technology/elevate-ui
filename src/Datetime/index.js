// @flow
import React from 'react';
import withStyles from 'react-jss';
import classNames from 'classnames';
import ReactDatetime from 'react-datetime';

import Label from '../Label';
import Validation from '../Validation';

type Props = {
  classes: Object,
  className: string,
  field: Object, // needs flow-typed https://github.com/flowtype/flow-typed/issues/1903
  form: Object, // needs flow-typed https://github.com/flowtype/flow-typed/issues/1903
  id: string,
  label: string,
  theme: Object,
};

const Datetime = ({
  classes,
  className,
  field,
  form: { errors, setFieldValue, setFieldTouched, touched },
  id,
  label,
  theme,
  ...rest
}: Props) => (
  <div className={classNames(classes.scaffold, className)}>
    {label && <Label htmlFor={id}>{label}</Label>}
    <ReactDatetime
      className={classes.rdt}
      inputProps={{ id: id, className: classes.input, autoComplete: 'off' }}
      {...field} // contains name, value, onBlur, onChange
      onBlur={date => setFieldTouched(field.name, date)} // overrides `field.onBlur`
      onChange={date => setFieldValue(field.name, date)} // overrides `field.onChange`
      {...rest}
    />
    {touched[field.name] &&
      errors[field.name] && <Validation error={errors[field.name]} />}
  </div>
);

export default withStyles(theme => ({
  scaffold: {
    width: '100%',
    margin: '8px auto 16px',
  },
  input: {
    display: 'block',
    width: '100%',
    height: '40px',
    color: theme.typography.bodyColor,
    fontFamily: 'inherit',
    fontWeight: '400',
    fontSize: '16px',
    lineHeight: '20px',
    backgroundColor: theme.colors.white,
    backgroundImage: `url('data:image/svg+xml;utf8,<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path fill="${
      theme.colors.gray300
    }" d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"/></svg>')`,
    backgroundRepeat: 'no-repeat',
    backgroundSize: '24px 24px',
    backgroundPosition: 'right 8px center',
    border: `1px solid ${theme.colors.gray300}`,
    padding: '8px 36px 8px 12px',
    appearance: 'none', // Reset default inputs for iOS/etc.
    boxShadow: 'none', // Reset default inputs for mozilla

    '&:focus': {
      outline: 'none', // Disable default focus glow
      boxShadow: theme.globalBoxShadow, // Add back focus style
    },

    '&:disabled': {
      cursor: 'not-allowed',
    },
  },
  rdt: {
    '&.rdt': {
      position: 'relative',
    },
    '& .rdtPicker': {
      display: 'none',
      position: 'absolute',
      width: '250px',
      padding: '4px',
      marginTop: '6px',
      zIndex: '99999 !important',
      backgroundColor: theme.colors.white,
      border: `1px solid ${theme.colors.gray300}`,
      boxShadow: theme.globalBoxShadow,
    },
    '&.rdtOpen .rdtPicker': {
      display: 'block',
    },
    '&.rdtStatic .rdtPicker': {
      boxShadow: 'none',
      position: 'static',
    },
    '& .rdtPicker .rdtTimeToggle': {
      textAlign: 'center',
    },
    '& .rdtPicker table': {
      width: '100%',
      margin: '0',
    },
    '& .rdtPicker td, & .rdtPicker th': {
      verticalAlign: 'middle',
      textAlign: 'center',
      height: '28px',
    },
    '& .rdtPicker td': {
      cursor: 'pointer',
    },
    '& .rdtPicker td.rdtDay:hover, & .rdtPicker td.rdtHour:hover, & .rdtPicker td.rdtMinute:hover, & .rdtPicker td.rdtSecond:hover, & .rdtPicker .rdtTimeToggle:hover': {
      background: '#eeeeee',
      cursor: 'pointer',
    },
    '& .rdtPicker td.rdtOld, & .rdtPicker td.rdtNew,': {
      color: '#999999',
    },

    '& .rdtPicker td.rdtToday': {
      position: 'relative',
    },
    '& .rdtPicker td.rdtToday:before': {
      content: "''",
      display: 'inline-block',
      borderLeft: '7px solid transparent',
      borderBottom: `7px solid ${theme.colors.secondary}`,
      borderTopColor: 'rgba(0, 0, 0, 0.2)',
      position: 'absolute',
      bottom: '4px',
      right: '4px',
    },
    '& .rdtPicker td.rdtActive, & .rdtPicker td.rdtActive:hover': {
      backgroundColor: theme.colors.secondary,
      color: '#fff',
      textShadow: '0 -1px 0 rgba(0, 0, 0, 0.25)',
    },
    '& .rdtPicker td.rdtActive.rdtToday:before': {
      borderBottomColor: '#fff',
    },
    '& .rdtPicker td.rdtDisabled, & .rdtPicker td.rdtDisabled:hover, & .rdtPicker td span.rdtDisabled, & .rdtPicker td span.rdtDisabled:hover': {
      background: 'none',
      color: '#999999',
      cursor: 'not-allowed',
    },
    '& .rdtPicker td span.rdtOld': {
      color: '#999999',
    },
    '& .rdtPicker thead tr:first-child th': {
      cursor: 'pointer',
    },
    '& .rdtPicker thead tr:first-child th:hover': {
      background: '#eeeeee',
    },
    '& .rdtPicker .dow': {
      width: '14.2857%',
      borderBottom: 'none',
      cursor: 'default',
    },
    '.rdtPicker th.rdtSwitch': {
      width: '100px',
    },
    '& .rdtPicker th.rdtNext, & .rdtPicker th.rdtPrev': {
      fontSize: '24px',
      verticalAlign: 'top',
    },
    '& .rdtPrev span, & .rdtNext span': {
      display: 'block',
      userSelect: 'none',
    },
    '& .rdtPicker th.rdtDisabled, & .rdtPicker th.rdtDisabled:hover': {
      background: 'none',
      color: '#999999',
      cursor: 'not-allowed',
    },
    '& .rdtPicker tfoot': {
      borderTop: `1px solid #f9f9f9`,
      paddingTop: '4px',
    },
    '& .rdtPicker button': {
      border: 'none',
      background: 'none',
      cursor: 'pointer',
    },
    '& .rdtPicker button:hover': {
      backgroundColor: '#eee',
    },
    '& .rdtPicker thead button': {
      width: '100%',
      height: '100%',
    },
    '& td.rdtMonth, & td.rdtYear': {
      height: '50px',
      width: '25%',
      cursor: 'pointer',
    },
    '& td.rdtMonth:hover, & td.rdtYear:hover': {
      background: '#eee',
    },
    '& .rdtCounters': {
      display: 'inline-block',
    },

    '& .rdtCounters > div': {
      float: 'left',
    },
    '& .rdtCounter': {
      width: '40px',
      height: '100px',
    },
    '& .rdtCounterSeparator': {
      lineHeight: '100px',
    },
    '& .rdtCounter .rdtBtn': {
      height: '40%',
      lineHeight: '40px',
      cursor: 'pointer',
      display: 'block',
      userSelect: 'none',
    },
    '& .rdtCounter .rdtBtn:hover': {
      background: '#eee',
    },
    '& .rdtCounter .rdtCount': {
      height: '20%',
      fontSize: '1.2em',
    },
    '& .rdtMilli': {
      verticalAlign: 'middle',
      paddingLeft: '8px',
      width: '48px',
    },
    '& .rdtMilli input': {
      width: '100%',
      fontSize: '1.2em',
      marginTop: '37px',
    },
    '& .rdtTime td': {
      cursor: 'default',
    },
  },
}))(Datetime);
