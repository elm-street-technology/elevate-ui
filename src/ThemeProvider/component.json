{
  "description": "",
  "displayName": "ThemeProvider",
  "methods": [],
  "props": {
    "withReset": {
      "defaultValue": {
        "value": "true",
        "computed": false
      },
      "required": false
    }
  }
}