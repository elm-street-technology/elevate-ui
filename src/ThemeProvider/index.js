// @flow
import React from "react";
import injectSheet, { ThemeProvider as JSSThemeProvider } from "react-jss";
import merge from "lodash/merge";

const colors = {
  primary: "#f15953",
  primaryDark: "#d24f45",
  primaryGlow: "#d24f45", // todo
  secondary: "#55c3ba",
  secondaryDark: "#44b0a7",
  tertiary: "#2e7fc2",
  quaternary: "#ffc629",
  white: "#fff",
  gray050: "#fbfcfd",
  gray100: "#edf1f4",
  gray200: "#dde2e7",
  gray300: "#ccd2d8",
  gray400: "#b3bac1",
  gray500: "#888f96",
  gray600: "#60686f",
  gray700: "#4b565f",
  gray800: "#5a5b5c",
  gray900: "#232c35",
  black: "#11181e",
  error: "#9c2929",
  facebook: "#3b5998",
  google: "#db4437",
  linkedin: "#007bb5",
  pinterest: "#bd081c",
  twitter: "#1da1f2",
};

const alertColors = {
  error: {
    color: "#9c2929",
    backgroundColor: "#fcf2f2",
    borderColor: "#f6dddd",
  },
  info: {
    color: "#0c5460",
    backgroundColor: "#d1ecf1",
    borderColor: "#bee5eb",
  },
  success: {
    color: "#246761",
    backgroundColor: "#ecf8f7",
    borderColor: "#d2e8e6",
  },
  warning: {
    color: "#856404",
    backgroundColor: "#fff3cd",
    borderColor: "#ffeeba",
  },
};

const typography = {
  bodyColor: "#2E2E35",
  fontFamily:
    "'Open Sans', -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif",
  fontSize: "1rem",
};

const transitions = {
  default: "all 200ms linear",
};

const breakpoints = (minWidth: number) => {
  if (!minWidth) {
    return new Error("Must enter a valid minWidth to use theme breakpoints");
  }
  return `@media (min-width: ${minWidth}px)`;
};

const globalBorderRadius = "4px";
const globalBoxShadow =
  "0 2px 5px 0 rgba(0,0,0,0.16), 0 2px 10px 0 rgba(0,0,0,0.12)";
const globalPadding = {
  padding: "12px",
  [breakpoints.tabletPortrait]: {
    padding: "16px",
  },
  [breakpoints.tabletLandscape]: {
    padding: "20px",
  },
  [breakpoints.desktop]: {
    padding: "24px",
  },
};

const shadowKeyUmbraOpacity = 0.2;
const shadowKeyPenumbraOpacity = 0.14;
const shadowAmbientShadowOpacity = 0.12;

function createShadow(...px) {
  return [
    `${px[0]}px ${px[1]}px ${px[2]}px ${
      px[3]
    }px rgba(0, 0, 0, ${shadowKeyUmbraOpacity})`,
    `${px[4]}px ${px[5]}px ${px[6]}px ${
      px[7]
    }px rgba(0, 0, 0, ${shadowKeyPenumbraOpacity})`,
    `${px[8]}px ${px[9]}px ${px[10]}px ${
      px[11]
    }px rgba(0, 0, 0, ${shadowAmbientShadowOpacity})`,
  ].join(",");
}

const shadows = [
  createShadow(0, 1, 3, 0, 0, 1, 1, 0, 0, 2, 1, -1),
  createShadow(0, 1, 5, 0, 0, 2, 2, 0, 0, 3, 1, -2),
  createShadow(0, 1, 8, 0, 0, 3, 4, 0, 0, 3, 3, -2),
  createShadow(0, 2, 4, -1, 0, 4, 5, 0, 0, 1, 10, 0),
  createShadow(0, 3, 5, -1, 0, 5, 8, 0, 0, 1, 14, 0),
  createShadow(0, 3, 5, -1, 0, 6, 10, 0, 0, 1, 18, 0),
  createShadow(0, 4, 5, -2, 0, 7, 10, 1, 0, 2, 16, 1),
  createShadow(0, 5, 5, -3, 0, 8, 10, 1, 0, 3, 14, 2),
  createShadow(0, 5, 6, -3, 0, 9, 12, 1, 0, 3, 16, 2),
  createShadow(0, 6, 6, -3, 0, 10, 14, 1, 0, 4, 18, 3),
  createShadow(0, 6, 7, -4, 0, 11, 15, 1, 0, 4, 20, 3),
  createShadow(0, 7, 8, -4, 0, 12, 17, 2, 0, 5, 22, 4),
  createShadow(0, 7, 8, -4, 0, 13, 19, 2, 0, 5, 24, 4),
  createShadow(0, 7, 9, -4, 0, 14, 21, 2, 0, 5, 26, 4),
  createShadow(0, 8, 9, -5, 0, 15, 22, 2, 0, 6, 28, 5),
  createShadow(0, 8, 10, -5, 0, 16, 24, 2, 0, 6, 30, 5),
  createShadow(0, 8, 11, -5, 0, 17, 26, 2, 0, 6, 32, 5),
  createShadow(0, 9, 11, -5, 0, 18, 28, 2, 0, 7, 34, 6),
  createShadow(0, 9, 12, -6, 0, 19, 29, 2, 0, 7, 36, 6),
  createShadow(0, 10, 13, -6, 0, 20, 31, 3, 0, 8, 38, 7),
  createShadow(0, 10, 13, -6, 0, 21, 33, 3, 0, 8, 40, 7),
  createShadow(0, 10, 14, -6, 0, 22, 35, 3, 0, 8, 42, 7),
  createShadow(0, 11, 14, -7, 0, 23, 36, 3, 0, 9, 44, 8),
  createShadow(0, 11, 15, -7, 0, 24, 38, 3, 0, 9, 46, 8),
];

const zIndex = {
  buttonBase: 0,
  buttonChildren: 1,
  selectDropdown: 2,
  tableLoading: -1,
  tableLoadingActive: 2,
  tableResizer: 10,
  tooltip: 1000,
  modal: 2000,
};

const defaultTheme = {
  alertColors,
  breakpoints,
  colors,
  globalBorderRadius,
  globalBoxShadow,
  globalPadding,
  shadows,
  transitions,
  typography,
  zIndex,
};

const GlobalsAndReset = injectSheet((theme) => ({
  "@global": {
    /* Eric Meyer Reset v2.0 */
    /*Â https://meyerweb.com/eric/tools/css/reset/ */
    "html, body, div, span, applet, object, iframe, h1, h2, h3, h4, h5, h6, p, blockquote, pre, a, abbr, acronym, address, big, cite, code, del, dfn, em, img, ins, kbd, q, s, samp, small, strike, strong, sub, sup, tt, var, b, u, i, center, dl, dt, dd, ol, ul, li, fieldset, form, label, legend, table, caption, tbody, tfoot, thead, tr, th, td, article, aside, canvas, details, embed, figure, figcaption, footer, header, hgroup, menu, nav, output, ruby, section, summary, time, mark, audio, video": {
      margin: "0",
      padding: "0",
      border: "0",
      fontSize: "100%",
      font: "inherit",
      verticalAlign: "baseline",
    },
    "article, aside, details, figcaption, figure, footer, header, hgroup, menu, nav, section": {
      display: "block",
    },
    body: {
      lineHeight: "1",
    },
    "ol, ul": {
      listStyle: "none",
    },
    "blockquote, q": {
      quotes: "none",
    },
    "blockquote:before, blockquote:after, q:before, q:after": {
      content: "''",
      content: "none", // eslint-disable-line
    },
    table: {
      borderCollapse: "collapse",
      borderSpacing: "0",
    },
    /* Globals */
    "html, body": {
      width: "100%",
      minHeight: "100%",
      fontFamily: theme.typography.fontFamily || "sans-serif",
      fontSize: theme.typography.fontSize || "1rem",
      fontWeight: "500",
      color: theme.typography.bodyColor || "#000",
      backgroundColor: "#fafafa",
      "-webkit-font-smoothing": "antialiased",
      "-moz-osx-font-smoothing": "grayscale",
      margin: "0",
      padding: "0",
    },
    /* Box-sizing border-box */
    /* https://www.paulirish.com/2012/box-sizing-border-box-ftw/ */
    html: {
      boxSizing: "border-box",
      background: "transparent",
    },
    "*, *:before, *:after": {
      boxSizing: "inherit",
    },
    /* Reset `button` to nothing */
    /* https://gist.github.com/chrisheninger/a860f87ef4e529b2df606768b97665a8 */
    button: {
      display: "inline-block",
      "-webkit-appearance": "none",
      "-moz-appearance": "none",
      userSelect: "none",
      textDecoration: "none",
      fontFamily: "inherit",
      fontSize: "inherit",
      color: "inherit",
      cursor: "pointer",
      border: "none",
      backgroundColor: "transparent",
      borderColor: "transparent",
      padding: "0",
      margin: "0",
      "&:disabled": {
        opacity: 0.6,
        cursor: "not-allowed",
      },
    },
  },
}))(({ children }) => children);

const ThemeProvider = ({ children, theme, withReset = true, ...rest }: any) => {
  const mergedTheme = theme ? merge(defaultTheme, theme) : defaultTheme;
  return (
    <JSSThemeProvider theme={mergedTheme} {...rest}>
      {withReset ? <GlobalsAndReset>{children}</GlobalsAndReset> : children}
    </JSSThemeProvider>
  );
};

export default ThemeProvider;
