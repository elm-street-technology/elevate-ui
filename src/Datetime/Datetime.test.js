import React from 'react';
import renderer from 'react-test-renderer';

import ThemeProvider from '../ThemeProvider';
import Datetime from './';

const WrappedDatetime = props => (
  <ThemeProvider>
    <Datetime
      field={{
        name: '',
        onChange: () => {},
        onBlur: () => {},
        value: new Date('04/29/2013 12:00 AM'),
      }}
      form={{
        errors: [],
        setFieldValue: () => {},
        setFieldTouched: () => {},
        touched: false,
      }}
      {...props}
    />
  </ThemeProvider>
);

test('renders without crashing, matches the snapshot', () => {
  const component = renderer.create(<WrappedDatetime id="startDate" />);
  const componentJSON = component.toJSON();
  expect(componentJSON).toMatchSnapshot();
});
