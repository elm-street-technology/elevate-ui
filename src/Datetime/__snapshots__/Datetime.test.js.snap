// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing, matches the snapshot 1`] = `
<div
  className="Datetime-scaffold-0-1-1"
>
  <div
    className="rdt Datetime-rdt-0-1-3"
  >
    <input
      autoComplete="off"
      className="Datetime-input-0-1-2"
      id="startDate"
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      type="text"
      value="04/29/2013 12:00 AM"
    />
    <div
      className="rdtPicker"
    >
      <div
        className="rdtDays"
      >
        <table>
          <thead>
            <tr>
              <th
                className="rdtPrev"
                onClick={[Function]}
              >
                <span>
                  ‹
                </span>
              </th>
              <th
                className="rdtSwitch"
                colSpan={5}
                data-value={3}
                onClick={[Function]}
              >
                April 2013
              </th>
              <th
                className="rdtNext"
                onClick={[Function]}
              >
                <span>
                  ›
                </span>
              </th>
            </tr>
            <tr>
              <th
                className="dow"
              >
                Su
              </th>
              <th
                className="dow"
              >
                Mo
              </th>
              <th
                className="dow"
              >
                Tu
              </th>
              <th
                className="dow"
              >
                We
              </th>
              <th
                className="dow"
              >
                Th
              </th>
              <th
                className="dow"
              >
                Fr
              </th>
              <th
                className="dow"
              >
                Sa
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                className="rdtDay rdtOld"
                data-value={31}
                onClick={[Function]}
              >
                31
              </td>
              <td
                className="rdtDay"
                data-value={1}
                onClick={[Function]}
              >
                1
              </td>
              <td
                className="rdtDay"
                data-value={2}
                onClick={[Function]}
              >
                2
              </td>
              <td
                className="rdtDay"
                data-value={3}
                onClick={[Function]}
              >
                3
              </td>
              <td
                className="rdtDay"
                data-value={4}
                onClick={[Function]}
              >
                4
              </td>
              <td
                className="rdtDay"
                data-value={5}
                onClick={[Function]}
              >
                5
              </td>
              <td
                className="rdtDay"
                data-value={6}
                onClick={[Function]}
              >
                6
              </td>
            </tr>
            <tr>
              <td
                className="rdtDay"
                data-value={7}
                onClick={[Function]}
              >
                7
              </td>
              <td
                className="rdtDay"
                data-value={8}
                onClick={[Function]}
              >
                8
              </td>
              <td
                className="rdtDay"
                data-value={9}
                onClick={[Function]}
              >
                9
              </td>
              <td
                className="rdtDay"
                data-value={10}
                onClick={[Function]}
              >
                10
              </td>
              <td
                className="rdtDay"
                data-value={11}
                onClick={[Function]}
              >
                11
              </td>
              <td
                className="rdtDay"
                data-value={12}
                onClick={[Function]}
              >
                12
              </td>
              <td
                className="rdtDay"
                data-value={13}
                onClick={[Function]}
              >
                13
              </td>
            </tr>
            <tr>
              <td
                className="rdtDay"
                data-value={14}
                onClick={[Function]}
              >
                14
              </td>
              <td
                className="rdtDay"
                data-value={15}
                onClick={[Function]}
              >
                15
              </td>
              <td
                className="rdtDay"
                data-value={16}
                onClick={[Function]}
              >
                16
              </td>
              <td
                className="rdtDay"
                data-value={17}
                onClick={[Function]}
              >
                17
              </td>
              <td
                className="rdtDay"
                data-value={18}
                onClick={[Function]}
              >
                18
              </td>
              <td
                className="rdtDay"
                data-value={19}
                onClick={[Function]}
              >
                19
              </td>
              <td
                className="rdtDay"
                data-value={20}
                onClick={[Function]}
              >
                20
              </td>
            </tr>
            <tr>
              <td
                className="rdtDay"
                data-value={21}
                onClick={[Function]}
              >
                21
              </td>
              <td
                className="rdtDay"
                data-value={22}
                onClick={[Function]}
              >
                22
              </td>
              <td
                className="rdtDay"
                data-value={23}
                onClick={[Function]}
              >
                23
              </td>
              <td
                className="rdtDay"
                data-value={24}
                onClick={[Function]}
              >
                24
              </td>
              <td
                className="rdtDay"
                data-value={25}
                onClick={[Function]}
              >
                25
              </td>
              <td
                className="rdtDay"
                data-value={26}
                onClick={[Function]}
              >
                26
              </td>
              <td
                className="rdtDay"
                data-value={27}
                onClick={[Function]}
              >
                27
              </td>
            </tr>
            <tr>
              <td
                className="rdtDay"
                data-value={28}
                onClick={[Function]}
              >
                28
              </td>
              <td
                className="rdtDay rdtActive"
                data-value={29}
                onClick={[Function]}
              >
                29
              </td>
              <td
                className="rdtDay"
                data-value={30}
                onClick={[Function]}
              >
                30
              </td>
              <td
                className="rdtDay rdtNew"
                data-value={1}
                onClick={[Function]}
              >
                1
              </td>
              <td
                className="rdtDay rdtNew"
                data-value={2}
                onClick={[Function]}
              >
                2
              </td>
              <td
                className="rdtDay rdtNew"
                data-value={3}
                onClick={[Function]}
              >
                3
              </td>
              <td
                className="rdtDay rdtNew"
                data-value={4}
                onClick={[Function]}
              >
                4
              </td>
            </tr>
            <tr>
              <td
                className="rdtDay rdtNew"
                data-value={5}
                onClick={[Function]}
              >
                5
              </td>
              <td
                className="rdtDay rdtNew"
                data-value={6}
                onClick={[Function]}
              >
                6
              </td>
              <td
                className="rdtDay rdtNew"
                data-value={7}
                onClick={[Function]}
              >
                7
              </td>
              <td
                className="rdtDay rdtNew"
                data-value={8}
                onClick={[Function]}
              >
                8
              </td>
              <td
                className="rdtDay rdtNew"
                data-value={9}
                onClick={[Function]}
              >
                9
              </td>
              <td
                className="rdtDay rdtNew"
                data-value={10}
                onClick={[Function]}
              >
                10
              </td>
              <td
                className="rdtDay rdtNew"
                data-value={11}
                onClick={[Function]}
              >
                11
              </td>
            </tr>
          </tbody>
          <tfoot>
            <tr>
              <td
                className="rdtTimeToggle"
                colSpan={7}
                onClick={[Function]}
              >
                12:00 AM
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
    </div>
  </div>
</div>
`;
